cpp/tests/strings/split_tests.cpp

```diff
@@ -308,6 +308,31 @@ TEST_F(StringsSplitTest, SplitRecordWhitespaceWithMaxSplit)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
 }
 
+TEST_F(StringsSplitTest, MultiByteDelimiters)
+{
+  // includes overlapping delimiters and delimiters that span across adjacent strings
+  auto input =
+    cudf::test::strings_column_wrapper({"u__", "w___x", "y____z", "{a=1}:{b=2}:", "{c=3}"});
+  auto view = cudf::strings_column_view(input);
+
+  auto result = cudf::strings::split_record(view, cudf::string_scalar("__"));
+  using LCW   = cudf::test::lists_column_wrapper<cudf::string_view>;
+  auto expected1 =
+    LCW({LCW{"u", ""}, LCW{"w", "_x"}, LCW{"y", "", "z"}, LCW{"{a=1}:{b=2}:"}, LCW{"{c=3}"}});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected1);
```

Comment:
Recommend putting each test into a block `{}` so you won't have to enumerate variables like `expected1`, `expected2` etc.

Reply:
I added additional tests for `split/rsplit` as well.